### PR TITLE
feat: cleanup platform libraries linkage

### PR DESCRIPTION
**Pull request by Mixaill**
Fixes possible issues with static sentry-native library.

References https://github.com/getsentry/sentry-native/issues/477

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -340,19 +345,29 @@ if(WIN32)
...
-	endif()
-	target_link_libraries(sentry PRIVATE dl)
-	target_link_libraries(sentry PRIVATE rt)
-elseif(ANDROID)
-	target_link_libraries(sentry PRIVATE dl log)
-elseif(MSVC)
-	target_link_libraries(sentry PRIVATE dbghelp version)
-elseif(MINGW)
-	target_link_libraries(sentry PRIVATE dbghelp version)
+#handle platform libraries
```
Comment (Swatinem) [inline CMakeLists.txt:348]:
```suggestion
# handle platform libraries
```

Please keep a leading space

File: README.md [hunk 2]
```diff
@@ -188,8 +188,8 @@ using `cmake -D BUILD_SHARED_LIBS=OFF ..`.
 - `SENTRY_BUILD_RUNTIMESTATIC` (Default: OFF):
   Enables linking with the static MSVC runtime. Has no effect if the compiler is not MSVC.
 
-- `SENTRY_LINK_PTHREAD` (Default: ON):
-  Links to the `pthread` library on unix targets.
+- `SENTRY_LINK_THREADS` (Default: ON):
```
Comment (Swatinem) [inline README.md:191]: I wonder if we should keep the old config name for backwards compatibility here.

Comment (Mixaill) [reply README.md:191]: Yes, I think we can. In 99% occurrences thread library will be `pthread`.

File: CMakeLists.txt (hunk 1 above)
Comment (Mixaill) [reply CMakeLists.txt]: Fixed

**Review by Swatinem — APPROVED** [review]

**Pull request merged**